lib/matplotlib/tests/test_scale.py

```diff
@@ -121,3 +121,21 @@ def test_logscale_nonpos_values():
 
     ax4.set_yscale('log')
     ax4.set_xscale('log')
+
+
+def test_invalid_log_lims():
+    # Check that invalid log scale limits are ignored
+    fig, ax = plt.subplots()
+    ax.scatter(range(0, 4), range(0, 4))
```

Comment:
Are you intentionally using data points that are not part of the valid log scale? I would have taken `range(1, 3)`.

Reply:
It's definitively meaningful. I just had to put in a bit of extra thought what the actual limit before `set_xlim` would be. So it's slightly more complicated than need be.

Change it or leave it as you wish.

Reply:
I think I'll leaveit.

Reply:
Would this fix #10782? In that case using negative values explicitely might make sense for a test (testing for 0 and a negative value independently, I mean).

Reply:
No, it doesn't. I think that bug is a "scatter markers don't do autoscaling" bug.